### PR TITLE
Lwm2m swmgmt rebase

### PR DESCRIPTION
**Pull request by DunderRoffe**

### CODE REVIEW

**Comment by DunderRoffe** [comment]
@JoshuaJeyaraj FYI. This is only the rebase ... I have not tested it

**Comment by DunderRoffe** [comment]
Tested by joshua

File: include/net/lwm2m.h [hunk 1]
```diff
@@ -311,6 +312,99 @@ struct coap_block_context *lwm2m_firmware_get_block_context();
 #endif
 #endif
 
+
+#if defined(CONFIG_LWM2M_SWMGMT_OBJ_SUPPORT)
+
+/**
+ * @breif Set callback to handle software activation requests
```
Comment (linusnielsen) [inline include/net/lwm2m.h:319]: It's "brief" :-)

Comment (linusnielsen) [reply include/net/lwm2m.h:319]: Normally, spelling mistakes are not a big deal, but in this case the @brief tag is used by tools to automatically produce documentation.

**Review by BassemMohsen — APPROVED** [review]

File: include/net/lwm2m.h (hunk 1 above)
Comment (JoshuaJeyaraj) [reply include/net/lwm2m.h:319]: I thought brief was some interpretation of brief in Swedish

**Review by linusnielsen — APPROVED** [review]

**Pull request merged**